src/core/utils/components/VirtualScroll.test.tsx

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { fireEvent, render } from '@testing-library/react';
+import React from 'react';
+import VirtualScroll from './VirtualScroll';
+
+const component = (
+  <div style={{ overflow: 'auto', height: 400 }} id='scroller'>
+    <VirtualScroll
+      itemsLength={1000}
+      itemRenderer={(index) => (
+        <div key={index} className='element' style={{ height: 40 }}>{`Element${
+          index + 1
+        }`}</div>
+      )}
+    />
+  </div>
+);
+
+const renderComponent = () => {
+  return render(component);
+};
+
+const renderEmptyComponent = () => {
+  return render(
+    <div style={{ overflow: 'auto', height: 400 }} id='scroller'>
+      <VirtualScroll
+        itemsLength={1000}
+        itemRenderer={(index) => <div key={index} className='element' />}
+      />
+    </div>,
+  );
+};
+
+beforeAll(() => {
+  // return correct values for container 'scroller' and children
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function getBoundingClientRectMock(this: Record<string, any>) {
+    if (Object.values(this)[0].memoizedProps.id === 'scroller') {
+      return { height: 400 } as DOMRect;
+    }
+    return { height: 40 } as DOMRect;
+  }
+  jest
+    .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+    .mockImplementation(getBoundingClientRectMock);
+});
+
+it('should render only few elements out of big list', () => {
+  const { container } = renderComponent();
+  expect(container.querySelectorAll('.element').length).toBe(30);
+  expect(container.querySelector('.element')?.textContent).toBe('Element1');
```

Comment:
Maybe also check what is the last element.

Reply:
added

Reply:
I though you will add last element check in all places you are checking first element too...

Reply:
added :D